spec.html

```diff
@@ -13504,9 +14433,11 @@ <h1>Runtime Semantics: Evaluation</h1>
             1. Let _lstr_ be ? ToString(_lprim_).
             1. Let _rstr_ be ? ToString(_rprim_).
             1. Return the string-concatenation of _lstr_ and _rstr_.
-          1. Let _lnum_ be ? ToNumber(_lprim_).
-          1. Let _rnum_ be ? ToNumber(_rprim_).
-          1. Return the result of applying the addition operation to _lnum_ and _rnum_. See the Note below <emu-xref href="#sec-applying-the-additive-operators-to-numbers"></emu-xref>.
+          1. Let _lnum_ be ? ToNumeric(_lprim_).
+          1. Let _rnum_ be ? ToNumeric(_rprim_).
+          1. If Type(_lnum_) is different from Type(_rnum_), throw a *TypeError* exception.
+          1. Let _T_ be Type(_lnum_).
+          1. Return _T_::add(_lnum_, _rnum_).
```

Comment:
this seems a bit strange and hand-wavey; is this notation something that's been used before? (ie, using the result of `Type()` as a dynamic reference to a thing?

Reply:
That particular notation isn't used in the current spec, but we do have something that's semantically similar, where Environment Record methods are "dynamically dispatched" based on the specific type of record. (Ditto Module Record methods.)

Reply:
I feel like the norm is to use slots, more like lnum.[[Add]](rnum)

Reply:
It is not very clear to me what steps i need to take here.

> I feel like the norm is to use slots, more like `lnum.[[Add]](rnum)`

@devsnek I don't remember seen places in spec where slot is used on Number primitives. Do you mind sharing where this is being used currently?

Reply:
@caiolima not on Number specifically, just in general. `object.[[Get]]()`, `module.[[Evaluate]]()`, etc.

Reply:
Currently, only objects have internal slots. I think it'd be pretty weird to extend that to numeric values.

For `module.[[Evaluate]]()`, you're probably thinking of `_module_.Evaluate()`, where `Evaluate` is a "concrete method" of module records. So far, only record types have "concrete methods", but I suppose we could extend that to numeric types.

Reply:
@jmdyck yes, thank you for the correction.

Reply:
still not convinced by this notation, but i don't have a better suggestion at this time.